src/Microsoft.AspNet.Diagnostics.Elm/Views/DetailsPage.cshtml

```diff
@@ -14,56 +14,89 @@
     }
 
     public DetailsPageModel Model { get; set; }
-}
 
-@helper LogRow(LogInfo log)
-{
-    if (log.Severity >= Model.Options.MinLevel &&
-        (string.IsNullOrEmpty(Model.Options.NamePrefix) || log.Name.StartsWith(Model.Options.NamePrefix, StringComparison.Ordinal)))
+    public HelperResult LogRow(LogInfo log)
     {
-        <tr>
-            <td>@string.Format("{0:MM/dd/yy}", log.Time)</td>
-            <td>@string.Format("{0:H:mm:ss}", log.Time)</td>
-            <td class="@log.Severity.ToString().ToLowerInvariant()">@log.Severity</td>
-            <td title="@log.Name">@log.Name</td>
-            <td title="@log.Message" class="logState" width="100px">@log.Message</td>
-            <td title="@log.Exception">@log.Exception</td>
-        </tr>
-    }
-}
-
-@helper Traverse(ScopeNode node)
-{
-    var messageIndex = 0;
-    var childIndex = 0;
-    while (messageIndex < node.Messages.Count && childIndex < node.Children.Count)
-    {
-        if (node.Messages[messageIndex].Time < node.Children[childIndex].StartTime)
-        {
-            @LogRow(node.Messages[messageIndex])
-            messageIndex++;
-        }
-        else
+        return new HelperResult((__razor_helper_writer) =>
         {
-            @Traverse(node.Children[childIndex])
-            childIndex++;
-        }
-    }
-    if (messageIndex < node.Messages.Count)
-    {
-        for (var i = messageIndex; i < node.Messages.Count; i++)
-        {
-            @LogRow(node.Messages[i])
+
+            if (log.Severity >= Model.Options.MinLevel &&
+                (string.IsNullOrEmpty(Model.Options.NamePrefix) || log.Name.StartsWith(Model.Options.NamePrefix, StringComparison.Ordinal)))
+            {
+
+                WriteLiteralTo(__razor_helper_writer, "        <tr>\r\n            <td>");
+                WriteTo(__razor_helper_writer, string.Format("{0:MM/dd/yy}", log.Time));
+                WriteLiteralTo(__razor_helper_writer, "</td>\r\n            <td>");
+                WriteTo(__razor_helper_writer, string.Format("{0:H:mm:ss}", log.Time));
+                WriteLiteralTo(__razor_helper_writer, "</td>\r\n            <td");
+                WriteAttributeTo(__razor_helper_writer, "class", Tuple.Create(" class=\"", 768), Tuple.Create("\"", 819),
```

Comment:
Is all this stuff just to work around the lack of `@helper`? All this `Tuple()` stuff is probably not needed, is it?